src/ast/mod.rs

```diff
@@ -457,6 +457,15 @@ pub enum Statement {
     /// ```
     /// Limitations: doesn't support `LIKE`,  `WHERE` or `SHOW COLUMNS FROM mytable FROM mydb;`
     ShowColumns { table_name: ObjectName },
+    /// SHOW <object>S
+    ///
+    /// ```sql
+    /// SHOW TABLES;
+    /// SHOW VIEWS;
+    /// SHOW SOURCES;
+    /// SHOW SINKS;
+    /// ```
+    Show { object_type: ObjectType },
```

Comment:
I think it's only top-level objects, i.e. not anything that can be part of a table. Right now it has four branches: Source, SInk, View, Table.

Reply:
"Object" is actually a SQL term. More specifically, "database object". See, for example, https://docs.oracle.com/cd/B19306_01/server.102/b14200/sql_elements007.htm. Basically, anything that is named and belongs to a database is a database object. So tables, views, sources, and sinks in our system, but also index, sequences, etc. in more complete RDBMSes. Not columns, though, because those belong to a table, not a database.

Long term I suspect we'll need to split this variant into pieces, though. `SHOW TABLES`, for example, is not example to `SHOW DATABASES.

Reply:
that makes sense, ty